flink-runtime/src/main/java/org/apache/flink/runtime/heartbeat/HeartbeatManagerSenderImpl.java

```diff
@@ -37,7 +37,7 @@
 
     private final long heartbeatPeriod;
 
-    HeartbeatManagerSenderImpl(
+    public HeartbeatManagerSenderImpl(
```

Comment:
I'm wondering whether it would be more reasonable to move `RecordingHeartbeatServices` into `org.apache.flink.runtime.heartbeat` instead of keeping it in `org.apache.flink.runtime.taskexecutor`. This way we wouldn't have to expose the constructors and keep everything around the heartbeat in a single package. I understand that this was already like that beforehand. But I'm curious what you think.

Reply:
Good idea, thanks! I moved it and also reduced the visibility of `HeartbeatManagerSenderImpl`.